.github/workflows/test.yml

```diff
@@ -11,152 +11,127 @@ on:
       - master
 
 jobs:
-  test_linux:
-    name: Run tests (Linux)
-    runs-on: ubuntu-20.04
+
+  test_with_bindgen:
+    runs-on: ${{ matrix.config.os }}
+
+    name: ${{ matrix.config.os }} (R-${{ matrix.config.r }} rust-${{ matrix.config.rust-version }})
+              
     strategy:
+      fail-fast: false
       matrix:
-        rust-version: [stable, beta, nightly]
+        config:
+          - {os: windows-latest, r: 'release', rust-version: 'stable-x86_64-pc-windows-msvc'}
+          - {os: windows-latest, r: 'release', rust-version: 'stable-i686-pc-windows-msvc'}
+          - {os: macOS-latest,   r: 'release', rust-version: 'stable'}
+          - {os: ubuntu-20.04,   r: 'release', rust-version: 'stable', rspm: "https://packagemanager.rstudio.com/cran/__linux__/focal/latest"}
+          - {os: ubuntu-20.04,   r: 'release', rust-version: 'nightly', rspm: "https://packagemanager.rstudio.com/cran/__linux__/focal/latest"}
+          # R-devel requires LD_LIBRARY_PATH
+          - {os: ubuntu-20.04,   r: 'devel',   rust-version: 'nightly', rspm: "https://packagemanager.rstudio.com/cran/__linux__/focal/latest"}   
```

Comment:
I think R devel should be tested with rust stable, so if something breaks we know whether it's R devel or rust nightly that is the problem. Also, could you expand the matrix so there is an R `oldrel` run for each OS/architecture? I'd like to have bindings for those versions as well (I realize this will break the test of precompiled bindings; don't worry about that or download the artifacts and add them, as you wish).

Reply:
Ok, we now test three versions of R for each architecture. I also added R-release rust-nightly generating no bindings to test rust features.

Actually, you were wrong about tests -- you (or someone else) implemented a fallback to default bindings if the exact version is missing, so tests were successful (and a warning was logged). I downloaded new bindings, now we have fallback, 4.0 release, 4.1 devel, and 3.6 oldrel versions for each target.